api/include/opentelemetry/context/runtime_context.h

```diff
@@ -108,15 +146,46 @@ class RuntimeContext
     return temp_context.GetValue(key);
   }
 
-protected:
-  // Provides a token with the passed in context
-  Token CreateToken(Context context) noexcept { return Token(context); }
+  /**
+   * Provide a custom runtime context storage.
+   *
+   * This provides a possibility to override the default thread-local runtime
+   * context storage. This has to be set before any spans are created by the
+   * application, otherwise context state information will be discarded, which
+   * might lead to unexpected results.
+   *
+   * @param storage a custom runtime context storage
+   */
+  static void SetRuntimeContextStorage(nostd::shared_ptr<RuntimeContextStorage> storage) noexcept
+  {
+    while (GetLock().test_and_set(std::memory_order_acquire))
```

Comment:
Wrap `GetLock().test_and_set()` and `GetLock().clear()` into an object ctor and dtor to make sure `clear()` is always called here?

Reply:
Good point. I'll wait for the result of the discussion whether we'll need this lock at all. If so, I can wrap the calls in some kind of lock guard.

Reply:
I removed the locks altogether.